tools/autograd/templates/Functions.cpp

```diff
@@ -425,6 +425,16 @@ Tensor cumsum_backward(const Tensor & x, int64_t dim) {
   return ret;
 }
 
+Tensor cummax_backward(const Tensor &grad, const Tensor &input, int64_t dim) {
```

Comment:
If you don't define a backward for this function in `derivatives.yaml`, you need to make sure it is differentiable.
In particular, you might want to use `at::scatter` instead of the native version to make sure you can get gradients for it.